### PR TITLE
Binance API Manager

### PR DESCRIPTION
**Pull request by porter707**
This branch is aimed to be a start on https://github.com/edeng23/binance-trade-bot/issues/118 by refactoring the Binance code into it's own class to make future improvements easier.

It currently depends on the logging class from my other branch https://github.com/edeng23/binance-trade-bot/pull/97 which just makes using the same logger across different python files easier, if you don't like the way that logging class is implemented I'm happy to implement any suggestions.

I've been trying to test this locally by manually calling the various functions (but not actually making any buys/sells) and it all appears ok, I'm waiting on  an actual buy/sell to happen which may be a while with the current dip.

### CODE REVIEW

File: binance_api_manager.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+from crypto_logger import Logger
+from models import Coin
+import math, requests, time
+
+
+class BinanceAPIManager:
+
+    APIKey = None
+    APISecret = None
+    Tld = None
```
Comment (rel-s) [inline binance_api_manager.py]: I don't think these members are necessary, they aren't really used anywhere outside __init__()

File: binance_api_manager.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    BinanceClient = None
+
+    def __init__(self, APIKey, APISecret, Tld):
+
+        self.APIKey = APIKey
+        self.APISecret = APISecret
+        self.Tld = Tld
```
Comment (rel-s) [inline binance_api_manager.py]: Same goes here, we can just pass these values on to Client() without saving them

**Comment by rel-s** [comment]
We would like to merge this PR soon, sorry for the wait! Could you rebase this? At the current state it's a bit hard to review as it's not entirely clear what changes are because of this PR and what are due to #97. Thanks!

**Comment by porter707** [comment]
@rel-s this is rebased now and ready for review

**Comment by edeng23** [comment]
@porter707 Thanks :) I'll run a few tests before merging.

File: binance_api_manager.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                )
+                time.sleep(1)
+            except BinanceAPIException as e:
+                self.logger.info(e)
+                time.sleep(2)
+            except Exception as e:
+                self.logger.info("Unexpected Error: {0}".format(e))
+
+        newbal = self.get_currency_balance(alt_symbol)
+        while newbal >= bal:
```
Comment (edeng23) [inline binance_api_manager.py]: I got this error after running your code:
2021-02-26 21:12:06,030 - crypto_trader_logger - INFO - Failed to Buy/Sell. Trying Again.
2021-02-26 21:12:06,032 - crypto_trader_logger - INFO - name 'bal' is not defined

I believe that "bal" should be replaced with "alt_balance" in accordance with @j-waters's revision.

Comment (porter707) [reply binance_api_manager.py]: Ahh yes, sorry thats fixed now

**Pull request merged**